### PR TITLE
Ab/java docs workflow definition

### PR DESCRIPTION
**Pull request by aarohib**
## What does this PR do?
Adds a new How to spawn a Child Workflow in Java as part of the archetype conformation effort.

### CODE REVIEW

File: docs/java/how-to-spawn-a-child-workflow-execution-in-java.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+### ParentClosePolicy
+
+When creating a Child Workflow, you can define a `ParentClosePolicy` that terminates, cancels, or abandons the Child Workflow Execution if the parent Workflow Execution stops.
+
+- `ABANDON`: When the parent stops, don't do anything with the Child Workflow.
+- `TERMINATE`: When the parent stops, terminate the Child Workflow
+- `REQUEST_CANCEL`: When the parent stops, terminate the Child Workflow
+
+You can set policies per Child Workflow, which means you can opt out of propagating terminations and cancellations on a per-child basis.
+This is useful for starting Child Workflows asynchronously, as shown in the following example:
```
Comment (flossypurse) [inline docs/java/how-to-spawn-a-child-workflow-execution-in-java.md:107]: This whole section of text is conceptual explanation which is covered in the "What is a Parent Close Policy" page: https://docs.temporal.io/docs/concepts/what-is-a-parent-close-policy

Additionally - this actually maps to "How to set a Parent Close Policy in Java" - which is done via Child Workflow Options i.e. "How to set `ChildWorkflowOptions` in Java" - not that you have to do all this here - just a heads up & fyi.

File: docs/java/how-to-spawn-a-child-workflow-execution-in-java.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+          Promise<String> greeting2 = Async.function(child2::composeGreeting, "Bye", name);
+
+          // Do something else here.
+          ...
+          return "First: " + greeting1.get() + ", second: " + greeting2.get();
+      }
+  }
+  ```
+
+- Send a Signal to a Child Workflow from the parent:
```
Comment (flossypurse) [inline docs/java/how-to-spawn-a-child-workflow-execution-in-java.md:70]: How to send a Signal to a Workflow Execution in Java

**Review by flossypurse — APPROVED** [review]

File: docs/java/how-to-spawn-a-child-workflow-execution-in-java.md (hunk 1 above)
Comment (aarohib) [reply docs/java/how-to-spawn-a-child-workflow-execution-in-java.md:107]: I'll do it. This is important.

**Comment by aarohib** [comment]
Closing this PR and addressing the content updates from comments here in a new one.

**Pull request merged**